### ISSUE TITLE
Support for EWKB?

### ISSUE THREAD

**Issue by tsauerwein** [geomet/geomet#28]
First, thanks for this great library! I have been using Shapely to convert from/to WKB, but I can't use it for geometries with 4 dimensions. So, I was quite happy to find this project! :+1: 

But it seems that EWKB is not supported? Or am I missing an undocumented option?

**Comment by larsbutler** [comment]
You're welcome. That use case specifically is actually why I started this project. :)

Which parts of EWKT/EWKB are you thinking about? Are you talking about the SRIDs, or something else? To be honest, I don't recall what the fundamental difference is between WKT/WKB and EWKT/EWKB. It's been awhile since I've looked at the specs.

**Comment by tsauerwein** [comment]
Yes, EWKB/EWKT includes the SRID. I am using geomet together with [GeoAlchemy](https://github.com/geoalchemy/geoalchemy2), and GeoAlchemy by default loads geometries as EWKB. It is possible to force GeoAlchemy to use WKB instead, so it is not really a problem in my case.

**Comment by larsbutler** [comment]
@tsauerwein 

I think the main problem is that GeoJSON doesn't have anything in its specification to express/store SRIDs. The intention of GeoMet is to provide lossless conversions (as much as possible, anyway) between WKT/WKB/GeoJSON.

It may be possible by injecting some "extra" attributes into the GeoJSON structure. Just throwing out ideas. For example:

`SRID=4326;POINT(-44.3 60.1)`

In GeoJSON, this could be:

```
{
  "type": "Point",
  "coordinates": [-44.3, 60.1],
  "meta": {
    "srid": "4326"
  }
}
```

There's no provision in the new [GeoJSON standard](https://tools.ietf.org/html/rfc7946)--as far as I catn tell--for such extensions, but this might be an option.

Perhaps I can summon @sgillies and get his opinion. Sean, any thoughts? :)

**Comment by larsbutler** [comment]
Ah, sorry, and now I just saw it: You can extend GeoJSON objects with [`properties`](https://tools.ietf.org/html/rfc7946#section-6).

**Comment by sgillies** [comment]
Geometry objects don't have a properties member, so I recommend add a "srid" member alongside "type" and "coordinates", like

```json
{
  "type": "Point",
  "coordinates": [-44.3, 60.1],
  "srid": "4326"
}
```

**Comment by larsbutler** [comment]
@tsauerwein Aside from SRID, what else would you expect to see? As far as I can tell, EWKT adds the SRID and "M" type geometries (like `POINTM(1 2 3)` or `LINESTRINGM(....)`). Is there anything else? This is about all I could find on EWKT: http://postgis.net/docs/using_postgis_dbmanagement.html#EWKB_EWKT.

Is this feature something you'd like to contribute/work on? I could probably get it started (and support something simple like `POINT`), but I don't have a lot of spare time right now and would be happy to accept a contribution.

**Comment by larsbutler** [comment]
@sgillies Ah, I see. If it's acceptable to add new arbitrary keys to the top level geometry object, couldn't one just add `properties`? Or does that violate the spec somehow? I haven't read through it in detail yet--just skimming so far.

**Comment by sgillies** [comment]
@larsbutler yes, it does violate RFC 7946. We've disallowed "properties" on Geometry and FeatureCollection objects to make parsing optimizations possible.

**Comment by larsbutler** [comment]
@sgillies If `properties` is disallowed, what about something else like `meta`?

```json
{
  "type": "Point",
  "coordinates": [-44.3, 60.1],
  "meta": {
    "srid": "4326",
    "foo": "bar"
  }
}
```
In other words is `properties` disallowed because of the name, or is it disallowed because it's an object (potentially with nested objects inside)?

**Comment by sgillies** [comment]
@larsbutler because of the name. "meta" is fine, though you might want to check around the web for other implementations using the same name, but since there's only one item to go in "meta", you may not need it.

**Comment by larsbutler** [comment]
@sgillies I couldn't find anything specific. I think `meta` makes more sense because it's extensible. For example, for this EWKT/EWKB feature, for `M` type geometries there would need to be some indicator. For example:

```json
{
  "type": "Point",
  "coordinates": [-44.3, 60.1, 100.0],
  "meta": {
    "srid": "4326",
    "components": "XYM" 
  }
}
```

Possible values of `components` are: ["XY", "XYZ", "XYM", XYZM"]. This is optional. Three dimensional geometries are assumed to be XYZ, unless specified.

@tsauerwein What do you think about that? We're kind of just making stuff up here, but I don't see any other way to preserve this information through format conversions.

**Comment by sgillies** [comment]
@larsbutler I think XYM coodinates will be problematic because the GeoJSON spec says that an optional third coordinate is Z: https://tools.ietf.org/html/rfc7946#section-3.1.1.

**Comment by larsbutler** [comment]
Fixed by #32.
